include/oneapi/dpl/pstl/hetero/algorithm_ranges_impl_hetero.h

```diff
@@ -1395,16 +1455,23 @@ __pattern_reduce_by_segment(__hetero_tag<_BackendTag> __tag, _ExecutionPolicy&&
         oneapi::dpl::__internal::__ranges::__pattern_walk_n(
             __tag, oneapi::dpl::__par_backend_hetero::make_wrapped_policy<__copy_keys_values_range_wrapper>(__exec),
             __copy_range,
-            oneapi::dpl::__ranges::zip_view(std::forward<_Range1>(__keys), std::forward<_Range2>(__values)),
-            oneapi::dpl::__ranges::zip_view(std::forward<_Range3>(__out_keys), std::forward<_Range4>(__out_values)));
+            oneapi::dpl::__ranges::zip_view(
+                oneapi::dpl::__ranges::__get_subscription_view(std::forward<_Range1>(__keys)),
```

Comment:
@dmitriy-sobolev I believe we need to call `__get_subscription_view` here because inside `zip_view` we have usages of `operator[]` from these ranges: https://github.com/uxlfoundation/oneDPL/blob/73cb69658be365dc82fe4c933d782d8b13444b80/include/oneapi/dpl/pstl/utils_ranges.h#L256